jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/startupchecks/CheckForDuplicatePaths.java

```diff
@@ -0,0 +1,159 @@
+package se.fortnox.reactivewizard.jaxrs.startupchecks;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.fortnox.reactivewizard.jaxrs.JaxRsResource;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+
+import javax.inject.Inject;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CheckForDuplicatePaths implements StartupCheck {
+
+    private final        StartupCheckConfig startupCheckConfig;
+    private static final Logger             LOG                = LoggerFactory.getLogger(CheckForDuplicatePaths.class);
+    public static final  Pattern            PATH_PARAM_PATTERN = Pattern.compile("\\{(\\w*)\\}");
+
+    @Inject
+    public CheckForDuplicatePaths(StartupCheckConfig startupCheckConfig) {
+        this.startupCheckConfig = startupCheckConfig;
+    }
+
+    @Override
+    public void check(List<JaxRsResource> resources) {
+        resources.forEach(jaxRsResource -> {
+            final Optional<JaxRsResource> duplicate = resources.stream()
+                .filter(res -> res != jaxRsResource && pathParamCollides(res, jaxRsResource))
```

Comment:
You will be comparing both (a, b) and (b, a) here. Not a big problem for production, but perhaps this code adds up to some seconds in your everyday work when you are restarting services locally and also in tests?